sklearn/decomposition/sparse_pca.py

```diff
@@ -66,6 +66,13 @@ class SparsePCA(BaseEstimator, TransformerMixin):
         If None, the random number generator is the RandomState instance used
         by `np.random`.
 
+    normalize_components : boolean
+        False : Use a version of Sparse PCA without components normalization
+        and without data centring. This is likely a bug and eventhough it's
+        the default for backward compatibility, this should not be used
```

Comment:
if `normalize_components=False` should not be used it but its the default for back compatibility, the usual deprecation cycle should be added

Reply:
my fault. You already have the deprecation warning. You should add when wash this added and when would it be removed, in  the docstring

Reply:
```py

Reply:
```py
        .. versionadded:: 0.20
        .. deprecated:: 0.22
   		``normalize_components`` was added and set to ``False`` for bakward compatibility. It would be set to ``True`` from 0.22 onwards.

Reply:
Now I'm not sure if we do this, so feel free to contradict me.

Reply:
Yes, technically we add the parameter, then change its value, then remove it. It's long!